plugins/settings/settings_plugin.c

```diff
@@ -360,17 +610,158 @@ settings_plugin_dbus_handle_set_enabled(
     return TRUE;
 }
 
+static
+gboolean
+settings_plugin_dbus_handle_get_all2(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed(self, call,
+        SETTINGS_ACTION_GET_ALL2, SETTINGS_DEFAULT_ACCESS_GET_ALL2)) {
+        org_sailfishos_nfc_settings_complete_get_all2(iface, call,
+            SETTINGS_DBUS_INTERFACE_VERSION, self->nfc_enabled,
+            settings_plugin_get_all_plugin_settings(self));
+    }
+    return TRUE;
+}
+
+static
+gboolean
+settings_plugin_dbus_handle_get_all_plugin_settings(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed(self, call,
+        SETTINGS_ACTION_GET_ALL_PLUGIN_SETTINGS,
+        SETTINGS_DEFAULT_ACCESS_GET_ALL_PLUGIN_SETTINGS)) {
+        org_sailfishos_nfc_settings_complete_get_all_plugin_settings(iface,
+            call, settings_plugin_get_all_plugin_settings(self));
+    }
+    return TRUE;
+}
+
+static
+gboolean
+settings_plugin_dbus_handle_get_plugin_settings(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    const char* plugin,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed(self, call,
+        SETTINGS_ACTION_GET_PLUGIN_SETTINGS,
+        SETTINGS_DEFAULT_ACCESS_GET_PLUGIN_SETTINGS)) {
+        const SettingsPluginConfig* pc = g_hash_table_lookup(self->plugins,
+            plugin);
+
+        if (pc) {
+            org_sailfishos_nfc_settings_complete_get_plugin_settings(iface,
+              call, settings_plugin_config_variant(pc->config));
+        } else {
+            g_dbus_method_invocation_return_error_literal(call,
+                SETTINGS_DBUS_ERROR, SETTINGS_ERROR_UNKNOWN_PLUGIN, plugin);
+        }
+    }
+    return TRUE;
+}
+
+static
+gboolean
+settings_plugin_dbus_handle_get_plugin_value(
+    OrgSailfishosNfcSettings* iface,
+    GDBusMethodInvocation* call,
+    const char* plugin,
+    const char* key,
+    gpointer user_data)
+{
+    SettingsPlugin* self = THIS(user_data);
+
+    if (settings_plugin_access_allowed(self, call,
```

Comment:
In case of not allowed: should it return access denied error or something?

Reply:
It completes the call with `AccessDenied`, will add a comment.